api/v1alpha3/virtualmachine_types.go

```diff
@@ -406,6 +406,14 @@ type VirtualMachineSpec struct {
 	// behavior. In other words, please be careful when choosing to upgrade a
 	// VM to a newer hardware version.
 	MinHardwareVersion int32 `json:"minHardwareVersion,omitempty"`
+
+	// BiosUUID describes the desired BIOS UUID for a VM.
+	// If omitted, this field defaults to a random UUID.
+	// When the bootstrap provider is Cloud-Init, this value is
+	// used as the Cloud-Init instance ID.
+	// +optional
+	// +kubebuilder:validation:Format:=uuid4
+	BiosUUID string `json:"biosUUID,omitempty"`
```

Comment:
I think this might break the B/R workflow for VMs using the cloud-init bootstrap provider. A new cloud-init instance ID is always required to ensure that the network configuration runs on the restored VM. If it's set from the `vm.spec.biosUUID`, the same value from the backup would be used in restored VMs and hence not trigger the network configuration.

Reply:
> A new cloud-init instance ID is always required

Existing code comment says same ID should be used?
https://github.com/vmware-tanzu/vm-operator/blob/df36cc7c94996357e5042cc1ff0d0feb6a818f9f/pkg/providers/vsphere/vmlifecycle/bootstrap_cloudinit.go#L53-L54

But I've yet to read up on details of the workflow. Where else would `InstanceIDAnnotation` be set and to what value?

Reply:
That comment about B/R is no longer applicable and should be removed. We explicitly delete this annotation during the restore to ensure a new cloud-init instance ID is set to the restored VM CR UID.  
The annotation is actually set during the upgrade process to the value of `iid-datasource-none`. This is to address an old bug when cloud-init was used sans userdata. See https://github.com/vmware-tanzu/vm-operator/pull/211 that initiated added this code in v1a1.

Reply:
Updated to use new `VirtualMachineBootstrapCloudInitSpec.InstanceID` field